src/scripts/main.js

```diff
@@ -1,3 +1,33 @@
 'use strict';
 
-// write your code here
+function formatNumber(num) {
+  return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+}
+
+const populationElements = document.querySelectorAll('.population');
+
+let totalPopulation = 0;
+let count = 0;
+
+populationElements.forEach((element) => {
+  const populationText = element.textContent.trim();
+  const populationNumber = parseInt(populationText.replace(/,/g, ''), 10);
+
+  if (!isNaN(populationNumber) && populationNumber > 0) {
+    totalPopulation += populationNumber;
+    count++;
+  }
+});
+
+const averagePopulation = count > 0 ? totalPopulation / count : 0;
```

Comment:
The calculation of the average population is correct, and the use of a ternary operator ensures that division by zero is avoided.